pinot-common/src/main/java/com/linkedin/pinot/common/config/ColumnPartitionConfig.java

```diff
@@ -114,4 +114,29 @@ public static String rangesToString(List<IntRange> ranges) {
     }
     return builder.toString();
   }
+
+  @Override
+  public boolean equals(Object o) {
+
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
```

Comment:
This is auto generated code by IDE. I think it is more stricter than instanceof because it does not allow to compare sub class with super class (and vice-versa).